blog/2024-07-10-uppy-4.0.md

```diff
@@ -0,0 +1,422 @@
+---
+title:
+  'New Uppy 4.0 major: TypeScript rewrite, Google Photos, React hooks, and much
+  more.'
+date: 2024-07-10
+authors: [aduh95, evgenia, mifi, murderlon]
+image: /img/blog/4.0/preview.jpg
+slug: 'uppy-4.0'
+published: true
+toc_max_heading_level: 2
+---
+
+<img src="/img/blog/4.0/preview.jpg" alt="Screenshot of Uppy Dashboad with text outlining the major new features in 4.0" />
+
+Hold on to your leashes, folks! Uppy 4.0 is here, and it’s more exciting than a
+tennis ball at the dog park! Our beloved Uppy mascot, the adorable coding
+canine, has been hard at work fetching all the latest updates for you.
+
+From a full TypeScript makeover to shiny new React hooks, and even Google Photos
+integration, this release is packed with treats that will make developers wag
+their tails in delight. So, sit, stay, and let’s dig into what makes Uppy 4.0
```

Comment:
Lmao this is all so confusing, who's a dog!
First we're saying that Uppy is a tennis ball at the dog park.
Then we're saying that Uppy is a dog.
Then we're saying that developers are dogs who wag their tails.

I'm pro this vibe, but I think we could be more focused on who is a dog in this scenario.

Reply:
I think the idea was that Uppy is a dog whose name is Uppy

Reply:
Agreed. The way I pretty much saw it when I helped conceive this idea is:

- Uppy, and to a certain extent the developers / blog writers behind it, can be attributed doglike qualities
- users of Uppy / readers of the blogs can be depicted as dog owners

I slightly changed up the intro in my PR. Couldn't help myself from adding a dumb pun. (In my mind, 'without further a-dog' = no more dog jokes, for now).